de.metas.printing/de.metas.printing.embedded-client/src/main/java/de/metas/printing/client/endpoint/LoopbackPrintConnectionEndpoint.java

```diff
@@ -136,10 +138,21 @@ public PrintPackage getNextPrintPackage()
 
 		final PRTCPrintPackageType xmlRequest = converterPRTCPrintPackageType.createPRTCPrintPackageTypeRequest(sessionId, transactionId);
 		final PRTCPrintPackageType xmlResponse = sendXmlObject(xmlRequest);
+		
 		if (xmlResponse == null)
 		{
 			return null;
 		}
+		else if (xmlResponse.getCPrintJobInstructionsID()!= null && xmlResponse.getCPrintJobInstructionsID().intValue()>0) 
+		{
+			// check if we deal with a pdf printing
+			final I_C_Print_Job_Instructions pji = InterfaceWrapperHelper.create(Env.getCtx(), xmlResponse.getCPrintJobInstructionsID().intValue(), I_C_Print_Job_Instructions.class, ITrx.TRXNAME_None);
+			// the HW printer is set in instruction only when the printer is PDF type
+			if (pji!= null && pji.getAD_PrinterHW_ID() > 0 )
```

Comment:
"the HW printer is set in instruction only when the printer is PDF type" 
might change without anyone telling this little method. I suggest to rather invoke an explicit printing service method to tell this method if it's PDF typ or no